### PR TITLE
Migrate FSTMutationTests to C++

### PR DESCRIPTION
**Pull request by wilhuff**
Migrate `FSTMutationTests` to `mutation_test.cc`, including all transform-related tests that weren't done previously.

Note the diffbase on #3568.

### CODE REVIEW

File: Firestore/core/test/firebase/firestore/model/mutation_test.cc [hunk 1]
```diff
@@ -128,98 +135,325 @@ TEST(MutationTest, PatchingDeletedDocumentsDoesNothing) {
...
+
+using TransformPairs = std::vector<std::pair<std::string, TransformOperation>>;
+
+// Helper to test a particular transform scenario.
+void TransformBaseDoc(const FieldValue::Map& base_data,
+                      TransformPairs transforms,
+                      const FieldValue::Map& expected_data) {
+  Document current_doc = Doc("collection/key", 0, base_data);
+
+  for (auto transform : transforms) {
```
Comment (rsgowman) [inline Firestore/core/test/firebase/firestore/model/mutation_test.cc]: `const auto&`?

File: Firestore/core/test/firebase/firestore/testutil/testutil.cc [hunk 2]
```diff
@@ -30,9 +30,12 @@ using model::DocumentComparator;
 using model::DocumentSet;
 using model::FieldMask;
 using model::FieldPath;
+using model::FieldTransform;
 using model::FieldValue;
 using model::ObjectValue;
 using model::Precondition;
+using model::TransformMutation;
```
Comment (rsgowman) [inline Firestore/core/test/firebase/firestore/testutil/testutil.cc]: I'm not sure this using statement is helpful. We've got another collision between the testutil::$Name and model::$Name. Currently, all usages of model::TransformMutation are qualified, and unqualified usages refer to the testutil helper.

File: Firestore/core/test/firebase/firestore/model/mutation_test.cc [hunk 3]
```diff
@@ -27,271 +27,236 @@
...
+const Timestamp now = Timestamp::Now();
+
+TEST(MutationTest, AppliesSetsToDocuments) {
+  Document base_doc =
+      Doc("collection/key", 0, Map("foo", "foo-value", "baz", "baz-value"));
+
+  Mutation set = SetMutation("collection/key", Map("bar", "bar-value"));
+  auto result = set.ApplyToLocalView(base_doc, base_doc, now);
+
+  ASSERT_NE(result, absl::nullopt);
```
[the remark this answers is not in the document]
Comment (wilhuff) [reply Firestore/core/test/firebase/firestore/model/mutation_test.cc]: Done.

File: Firestore/core/test/firebase/firestore/model/mutation_test.cc [hunk 4]
```diff
@@ -128,98 +135,325 @@ TEST(MutationTest, PatchingDeletedDocumentsDoesNothing) {
...
+  EXPECT_EQ(result, expected_doc);
+}
+
+namespace {
+
+using TransformPairs = std::vector<std::pair<std::string, TransformOperation>>;
+
+// Helper to test a particular transform scenario.
+void TransformBaseDoc(const FieldValue::Map& base_data,
+                      TransformPairs transforms,
```
[the remark this answers is not in the document]
Comment (wilhuff) [reply Firestore/core/test/firebase/firestore/model/mutation_test.cc]: Done.

File: Firestore/core/test/firebase/firestore/model/mutation_test.cc (hunk 1 above)
Comment (wilhuff) [reply Firestore/core/test/firebase/firestore/model/mutation_test.cc]: Done.

File: Firestore/core/test/firebase/firestore/testutil/testutil.cc (hunk 2 above)
Comment (wilhuff) [reply Firestore/core/test/firebase/firestore/testutil/testutil.cc]: Good call. Removed.

**Pull request merged**